scripts/analyze_deps.py

```diff
@@ -242,18 +242,32 @@ def render_report(output_path, report_context):
             sys.exit(0)
 
     frozen = {}
+    overrides = {}
+    override_count = 0
     try:
         with io.open(frozen_filename, 'r', encoding='utf-8-sig') as frozen_file:
             for line in frozen_file:
-                req_name, spec = parse_req(line)
-                frozen[req_name] = [spec]
+                if line.startswith('#override'):
+                    _, lib_name, req_override = line.split(' ', 2)
+                    record_dep(overrides, req_override, lib_name)
+                    override_count += 1
+                elif not line.startswith('#'):
+                    req_name, spec = parse_req(line)
+                    frozen[req_name] = [spec]
     except:
         print('Unable to open shared_requirements.txt, shared requirements have not been validated')
 
     missing_reqs, new_reqs, changed_reqs = {}, {}, {}
+    non_overridden_reqs_count = 0
     if frozen:
         flat_deps = {req: sorted(dependencies[req].keys()) for req in dependencies}
         missing_reqs, new_reqs, changed_reqs = dict_compare(frozen, flat_deps)
+        if args.verbose and len(overrides) > 0:
+            print('\nThe following requirement overrides are in place:')
+            for overridden_req in overrides:
```

Comment:
How does `for...in` work in python? Just enumerates the object's keys? If I'm reading Stack Overflow correctly, you may be able to say `for attr, value in overrides` and then `for spec in value` on the below line of code.

Or I could be completely wrong but this is fun!

Reply:
Yep, when the object is a dict, it enumerates the object's keys. You can use `mydict.items()` to enumerate through pairs of `(key, value)` instead if desired, but the semantics there changed between python 2 (it returns a copy of all the items, use `.iteritems()` to get an iterator) vs python 3 (it returns an iterator). I always try to write 2/3 clean code unless there is a specific thing I simply can't do, so I usually avoid using those functions and just iterate through the keys.